helix/appendix/alternate-visual-studio-structures.rst

```diff
@@ -0,0 +1,231 @@
+
+.. admonition:: Sitecore Habitat
+
+    Newer versions of `Sitecore Habitat <https://github.com/Sitecore/Habitat>`__
+    have an updated ``gulpfile`` which targets the entirety of the Visual Studio
+    Solution for builds.
+
+    .. code-block:: js
+
+        gulp.task("Publish-All-Projects",
+            function(callback) {
+                var msbuild = new _msbuild(callback);
+                msbuild.solutionName = config.solutionName + '.sln';
+                var overrideParams = [];
+                overrideParams.push('/p:Configuration=' + config.buildConfiguration);
+                overrideParams.push('/p:DeployOnBuild=true');
+                overrideParams.push('/p:DeployDefaultTarget=WebPublish');
+                overrideParams.push('/p:WebPublishMethod=FileSystem');
+                overrideParams.push('/p:DeleteExistingFiles=false');
+                overrideParams.push('/p:publishUrl=' + config.websiteRoot);
+                overrideParams.push('/m');
+                overrideParams.push('/restore');
+                msbuild.config('overrideParams', overrideParams);
+                msbuild.config('version', config.buildToolsVersion);
+                msbuild.build();
+            });
+
+    You can alternatively use Sitecore TDS or Helix Publishing Pipeline
+    for local deployments, as seen in the new Sitecore Helix Examples.
+
+
+Are there other tooling improvements which can help?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If your primary concern is the overhead of creating projects which follow
+the Sitecore Helix conventions (naming, filesystem structure, etc.), then
+you might consider some of the community tools which make this easier by
+providing project "templates" which save you some clicks. Take care when
+using these however that you remove any generated projects
+`which the module does not need <#does-your-solution-contain-projects-you-do-not-need>`__.
+
+To improve the performance of Visual Studio and reduce build times, you can
+also consider using the `Filtered Solutions <https://docs.microsoft.com/en-us/visualstudio/ide/filtered-solutions?view=vs-2019>`__
+feature of Visual Studio 2019 to reduce the number of projects open when working
+on a particular module.
+
+
+The role of Visual Studio projects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Aside from the pragmatic issues, the architectural question at the root of
+this decision is the role of a project/assembly in Visual Studio. This is
+essentially the subject of Robert C. Martin's *Principles of Package and Component Design*,
+which contain the `underlying principles <package-principles>`__ that Sitecore Helix
+is based on. Based on these principles, Visual Studio projects form a "physical" boundary
+and unit of change for modules in Sitecore Helix.
+
+Martin however makes an assumption that modules (packages) are released / deployed
+independently. Due to the versioning and testing complexity this would put on the typical Sitecore
```

Comment:
I know Joe Bissol used to do exactly this @ Comcast because it was unreasonable at their scale to do anything BUT deploy individual module updates.

Not that your average Sitecore dev need take this on. But it is a stronger boundary in a DDD sense, for (very large) teams that need such guarantees.

Reply:
Clarified language a bit to make the exception case for this recommendation apparent